cdc/log.cc

```diff
@@ -675,15 +675,102 @@ utils::UUID generate_timeuuid(api::timestamp_type t) {
+    const bytes _tuuid;
+    // The timestamp of the created log mutation cells
+    const api::timestamp_type _ts;
+    // The ttl of the created log mutation cells
+    const ttl_opt _ttl;
+
+    // Keeps the next cdc$batch_seq_no value
+    int _batch_no = 0;
+
+    // The mutation under construction
+    mutation& _log_mut;
+
+public:
+    log_mutation_builder(mutation& log_mut, api::timestamp_type ts,
+                         const partition_key& base_pk, const schema& base_schema)
+        : _base_schema(base_schema), _log_schema(*log_mut.schema()),
+          _op_col(*_log_schema.get_column_definition(log_meta_column_name_bytes("operation"))),
+          _ttl_col(*_log_schema.get_column_definition(log_meta_column_name_bytes("ttl"))),
+          _base_pk(base_pk),
+          _tuuid(timeuuid_type->decompose(generate_timeuuid(ts))),
+          _ts(ts),
+          _ttl(_base_schema.cdc_options().ttl()
+                  ? std::optional{std::chrono::seconds(_base_schema.cdc_options().ttl())} : std::nullopt),
+          _log_mut(log_mut)
+    {}
+
+    clustering_key allocate_new_log_row() {
+        const auto log_ck = clustering_key::from_exploded(_log_schema, { _tuuid, int32_type->decompose(_batch_no++) });
+        auto pk_value = _base_pk.explode(_base_schema);
+        size_t pos = 0;
+        for (const auto& column : _base_schema.partition_key_columns()) {
+            assert (pos < pk_value.size());
+
+            auto& cdef = *_log_schema.get_column_definition(log_data_column_name_bytes(column.name()));
+            _log_mut.set_cell(log_ck, cdef, atomic_cell::make_live(*column.type, _ts, bytes_view(pk_value[pos]), _ttl));
+            ++pos;
+        }
+        return log_ck;
+    }
+
+    void set_clustering_columns(const clustering_key& log_ck, const clustering_key_prefix& base_ckey) {
+        auto ck_value = base_ckey.explode(_log_schema);
+        size_t pos = 0;
+        for (const auto& column : _base_schema.clustering_key_columns()) {
+            if (pos >= ck_value.size()) {
+                break;
+            }
+            auto& cdef = *_log_schema.get_column_definition(log_data_column_name_bytes(column.name()));
+            _log_mut.set_cell(log_ck, cdef, atomic_cell::make_live(*column.type, _ts, bytes_view(ck_value[pos]), _ttl));
+            ++pos;
+        }
+    }
+
+    void set_operation(const clustering_key& log_ck, operation op) {
+        _log_mut.set_cell(log_ck, _op_col, atomic_cell::make_live(
+                    *_op_col.type, _ts, _op_col.type->decompose(operation_native_type(op)), _ttl));
+    }
+
+    void set_ttl(const clustering_key& log_ck, ttl_opt ttl) {
+        if (ttl) {
+            _log_mut.set_cell(log_ck, _ttl_col, atomic_cell::make_live(
+                        *_ttl_col.type, _ts, _ttl_col.type->decompose(ttl->count()), _ttl));
+        }
+    }
+
+    void set_value(const clustering_key& log_ck, const column_definition& base_cdef, bytes value) {
```

Comment:
How about `bytes_view` here and below?
I don't know the higher context - maybe it's too tedious to do, so I'm not insisting.